lib/chef/provider/package.rb

```diff
@@ -207,6 +209,15 @@ def action_reconfig
       end
 
       # @todo use composition rather than inheritance
+
+      def with_subclass_array_wrapper(name, version)
+        if supports_arrays?
+          yield [name].flatten, [version].flatten
+        else
+          yield name, version
```

Comment:
Is this a convenient generic place to raise for resource that don't support arrays when an array is passed? #supports_arrays? is new here, so it'd have to be added to all the other classes. Probably not worth it.